include/storage/io.hpp

```diff
@@ -0,0 +1,58 @@
+#ifndef OSRM_STORAGE_IO_HPP_
+#define OSRM_STORAGE_IO_HPP_
+
+#include "util/fingerprint.hpp"
+#include "util/simple_logger.hpp"
+
+#include <boost/filesystem/fstream.hpp>
+
+#include <tuple>
+
+namespace osrm
+{
+namespace storage
+{
+namespace io
+{
+
+// Returns the checksum and the number of nodes and number of edges
+std::tuple<unsigned, std::size_t, std::size_t> readHSGRSize(boost::filesystem::ifstream &input_stream)
```

Comment:
Maybe call it `readHSGRMetadata` (?) - also all of those functions will return a fingerprint and some additional data, right? What about making a namedtuple type instead so call sites don't have to `get<Nth>` all over the place.

    struct { Fingerprint fingerprint; size_t nbNodes; size_t nbEdges; }

or similar.